dotnet/Gherkin/CucumberMessages/AstMessagesConverter.cs

```diff
@@ -74,6 +74,9 @@ private Feature ConvertFeature(Ast.GherkinDocument gherkinDocument)
 
     private static Location ConvertLocation(Ast.Location location)
     {
+        // In Reqnroll and SpecFlow, plugins may provide dynamically created Example data rows that don't have source locations. Hard-coding these to (0, 0).
+        if (location == null)

```

Comment:
This sounds like a leaky abstraction. Shouldn't the callers of this method perform this operation? Gherkin itself does not have a concept it dynamically created example rows.